src/infra/dataverse/sparql/sparqlGateway.ts

```diff
@@ -0,0 +1,142 @@
+      PREFIX datasetMetadata: <https://ontology.okp4.space/metadata/dataset/>
+      PREFIX dataspaceMetadata: <https://ontology.okp4.space/metadata/dataspace/>
+      SELECT ?id ?metadata ?title ?description ?type
+      WHERE {
+        {
+          ?id rdf:type core:Service .
+          ?metadata rdf:type serviceMetadata:GeneralMetadata .
+        } UNION {
+          ?id rdf:type core:Dataset .
+          ?metadata rdf:type datasetMetadata:GeneralMetadata .
+        } UNION {          
+          ?id rdf:type core:DataSpace .
+          ?metadata rdf:type dataspaceMetadata:GeneralMetadata .
+        }
+        ${filters.byType !== 'all' ? `FILTER ( contains(str(?type), "${filters.byType}" ))` : ''}
+        ?id rdf:type ?type .
+        ?type rdf:type owl:Class .
+        ?metadata core:describes ?id .
+        ?metadata core:hasTitle ?title .
+        ?metadata core:hasDescription ?description .
+        FILTER ( langMatches(lang(?title),'${language}') && langMatches(lang(?description),'${language}') )
+      }
+      LIMIT ${limit}
+      OFFSET ${offset}
+`
+
+    const fetchHeaders: RequestInit = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+        Authorization: `Basic ${APP_ENV.sparql['credentials']}`,
+        accept: 'application/sparql-results+json'
+      },
+      body: `query=${encodeURIComponent(query)}`
+    }
+
+    const fetchDataverse = (): TE.TaskEither<Error, Response> =>
+      TE.tryCatch(
+        async () => fetch(APP_ENV.sparql['endpoint'], fetchHeaders),
+        reason =>
+          reason instanceof Error
+            ? reason
+            : new Error(`Oops.. Something went wrong fetching ontology: ${JSON.stringify(reason)}`)
+      )
+
+    const serializeResponse = (response: Response): TE.TaskEither<Error, SparqlResult> =>
+      TE.tryCatch(
+        async () => response.json(),
+        reason =>
+          reason instanceof Error
+            ? reason
+            : new Error(
+                `Oops.. Something went wrong serializing sparql response: ${JSON.stringify(reason)}`
+              )
+      )
+
+    const splitBindingId = (b: SparqlBinding): O.Option<[SparqlBinding, string]> =>
+      pipe(
+        O.fromNullable(b.id.value.split('/').at(-1)),
+        O.map(a => [b, a])
+      )
+
+    const splitBindingType = ([b, a]: [SparqlBinding, string]): O.Option<
+      [SparqlBinding, string, string]
+    > =>
+      pipe(
+        O.fromNullable(b.type.value.split('/').at(-1)),
+        O.map(c => [b, a, c])
+      )
+
+    const getDataverseEntityLengthOption = O.fromPredicate(
+      (dataverseEntity: DataverseEntity) => dataverseEntity.length === limit
+    )
+    const filterDataverseEntity = (dataverseEntity: DataverseEntity): DataverseEntity =>
+      pipe(
+        dataverseEntity,
+        getDataverseEntityLengthOption,
+        O.match(
+          () => dataverseEntity,
+          e => A.dropRight(1)(e)
+        )
+      )
```

Comment:
I want to remove the extra element of the entity, to store `limit - 1` elements in the array.
Here `limit` equals `21`, so I'll suggest to do this instead: 

```ts
const filterDataverseEntity = A.takeLeft(limit - 1)
```

Reply:
In fact, the limit should represent the maximum number of elements retrieved. I believe it would be more logical to set the limit to 20, for example, and then call the gateway with limit + 1 (21). This would create a more consistent approach.

Reply:
Make sense, I'll make the change.